tests/Earthfile

```diff
@@ -1269,6 +1270,25 @@ pipelines:
         --should_fail=true \
         --output_contains="valid triggers include"
 
+test-works-without-earthly-server:
+    # first warm-up the cache
+    DO +RUN_EARTHLY --verbose=false --earthfile=true.earth --target=+true
+
+    RUN echo "#!/bin/sh
+    set -e
+    unset EARTHLY_DISABLE_ANALYTICS
+    for host in ci.earthly.dev api.earthly.dev; do
+      echo \"blocking \$host\"
+      for ip in \$(getent hosts \$host | awk '{ print \$1 }'); do
+        iptables -A OUTPUT -d \$ip -j DROP
+      done
+    done
+    date +%s > start-time
+    " > /tmp/precommand && chmod +x /tmp/precommand
+    DO +RUN_EARTHLY --verbose=false --pre_command=/tmp/precommand --post_command=" && date +%s > end-time" --earthfile=true.earth --target=+true
+    RUN if grep "failed sending analytics" earthly.output; then echo "analytics should have failed silently, but didnt" && exit 1; fi
+    RUN export duration=$(echo "$(cat end-time) - $(cat start-time)" | bc) && test "$duration" -lt 45 || (echo "it took $duration seconds to run +true, which is too long" && exit 1)
```

Comment:
I really don't like this part -- it looks like we have a 30 second timeout that occurs if our server doesn't respond.

Reply:
CC @idodod